keel-core/src/main/kotlin/com/netflix/spinnaker/keel/notifications/Notifier.kt

```diff
@@ -0,0 +1,8 @@
+package com.netflix.spinnaker.keel.notifications
+
+/**
+ * All valid notifiers
+ */
+enum class Notifier {
```

Comment:
I am sure I wrote more comments than I can now find in the previous PR. I don't know what happened to them... 😞

Anyway, I had made a suggestion to refactor these models slightly to improve the naming and encapsulation. Something like this:
```kotlin
enum class NotificationType {
  RESOURCE_UNHEALTHY
}

abstract class Notification(
  abstract val scope: NotificationScope,
  abstract val ref: String,
  val subject: String,
  val body: String,
  val color: String = "#cccccc"
)

data class ResourceNotification(
  val resource: Resource<*>,
  val subject: String,
  val body: String,
  val color: String = "#cccccc"
) : Notification(NotificationScope.RESOURCE, resource.id, subject, body, color)

data class ResourceUnhealthyNotification(
  override val resource: Resource<*>
) : ResourceNotification(resource, "Managed Delivery", "The resource ${resource.id} is unhealthy.")
```
That way, your repository methods could just take the instance of the `Notification` object, which would encapsulate all the information you need to persist in the database. It would also remove the "notifier" term from these classes and leave that term for the classes which actually do the work of sending the notifications.

Reply:
P.S. This is not an original idea by any means. We use the same approach with the `PersistentEvent` and `ResourceEvent` class hierarchy. 🙂

Reply:
@luispollo I'm going to take some of these suggestions, but not all of them. I think what you're trying to convey is that some refactoring to the structures used here can help them be a lot more easily extendable. 

In the class description, I commented that this class will need work to make it super generic and that I don't think it's worth it at this moment since we only have one type of notification. It took me a long time to understand why this refactoring would be helpful, so in the future if you could add the "why" before suggesting a big change that would help me understand where you're coming from

Reply:
Thanks for being open to the suggestions, @emjburns. While extensibility comes as a nice side-effect of the suggested refactoring, my primary motivation as I tried to explain above was to improve encapsulation and naming. I've observed that we have occurrences in other places in keel as well where we tend to pass in discrete fields to a repository method when the goal is to persist an entire object, and so encapsulating those fields in the corresponding object is just good OO design, IMHO. This is why I suggested moving the notification type and the entity ID (`ref`) into the `Notification` class. That way your `NotificationEvent` could just carry a `Notification`, instead of the discrete fields, and similarly your repository methods would just take a `Notification` as a parameter.

The changes you made were great already. Up to you whether you think the additional bit described above should be done in this PR. I'd be happy to make those changes later as well if it helps.

Reply:
Discussed offline, for now keeping the three fields since we don't persist the message itself. In the future if we decide to save the message we will rethink this